src/ext/java/org/opentripplanner/ext/ridehailing/RideHailingDepartureTimeShifter.java

```diff
@@ -0,0 +1,132 @@
+package org.opentripplanner.ext.ridehailing;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import javax.annotation.Nonnull;
+import org.opentripplanner.ext.ridehailing.model.ArrivalTime;
+import org.opentripplanner.framework.geometry.WgsCoordinate;
+import org.opentripplanner.routing.api.request.RouteRequest;
+import org.opentripplanner.routing.api.request.StreetMode;
+import org.opentripplanner.routing.api.response.InputField;
+import org.opentripplanner.routing.api.response.RoutingError;
+import org.opentripplanner.routing.api.response.RoutingErrorCode;
+import org.opentripplanner.standalone.api.OtpServerRequestContext;
+import org.opentripplanner.transit.model.framework.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility method to shift the start of the journey to the earliest time that a vehicle can arrive.
+ */
+public class RideHailingDepartureTimeShifter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RideHailingDepartureTimeShifter.class);
+  /**
+   * When is a start time far enough in the future so that we don't need to check the service and
+   * simply presume that a vehicle can arrive on time.
```

Comment:
I do not think this is the correct way of handling this. It should be handled by inserting wait time in the beginning of the request instead. Modifing the request from the user feels wrong, and may have implication that is hard to see. Also, this is not consistent with how public transit is treated, if you stand on the platform we do NOT time-shift the departure to the first departure of the train... This will produce a cost witch is too low - hence in favor of RideHailing.

Reply:
That's a good point and I agree with this. Can you give me a hint on how to do this?

Reply:
So the way this is done in flex is this: https://github.com/opentripplanner/OpenTripPlanner/blob/cf5d59f5ac26e8a821c62abeaf2dc8d45145c6b2/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/router/TransitRouter.java#L227-L239

However, I don't feel like this is a good example of an interaction between the core and flex. 

Do you agree @t2gran?

Maybe I remove the time shifting for now and we have a design session on how to do it cleanly.

WDYT?